python/ray/llm/_internal/serve/deployments/llm/vllm/vllm_engine.py

```diff
@@ -567,7 +580,7 @@ def apply_hf_chat_template(model_config, **kwargs):
             "prompt": prompt_text,
             "prompt_token_ids": prompt_token_ids,
             "request_id": request_id,
-            "sampling_params": VLLMSamplingParams.from_prompt(prompt),
+            "sampling_params": self._parse_sampling_params(prompt),
```

Comment:
`VLLMGenerationRequest.sampling_params` type is not set to be `VLLMInternalSamplingParam`. It is set to be `VLLMSamplingParam` which can cause confusion later. Let's fix the type in `VLLMGenerationRequest`.